src/main/java/frc/robot/subsystems/drive/Drive.java

```diff
@@ -49,6 +85,15 @@ public void stop() {
         io.setVoltage(0.0, 0.0);
     }
 
+    public void aim() {
+        this.forward = 0;
+        this.rotation = targetHeading.getAsDouble();
```

Comment:
Here you would want the rotation controller that takes in a setpoint of 0 and a process variable of target heading

Naming this targetHeading is confusing because it's really headingError that you're representing

What you have here is fundamentally a P controller with gain of 1 (since your control output is simply the error itself). For an error in radians, you may need higher gain since a 0.1 radian error is large but will not result in much turn power

If it slows down early you may need to increase the P or I gain to get better results

Reply:
@jkleiber. Just ran the PID Controller with the heading error as measurement and 0 as setpoint. It resulted in the bot spinning in circles (although it did slow down minutely around the setpoint). Here is a graph (yellow is heading error; blue is the robot heading).
<img width="459" alt="Screenshot 2024-02-14 at 6 01 41 PM" src="https://github.com/team401/2024-everybot/assets/108961470/bb25a019-48f2-4f7a-bd39-10c7586927a8">

Reply:
I have fixed this by updating target calculation. Now it takes the target vector y and x and performs atan2. Without accounting for current heading, so it returns the actual target heading rather than error).

By adding this and updating pid controller to have process variable of current heading and set point as target heading, the controller works how it should. Now just need to update PID values to prevent over rotating.
updated picture:
<img width="195" alt="Screenshot 2024-02-14 at 6 31 50 PM" src="https://github.com/team401/2024-everybot/assets/108961470/96cf388d-6cf8-493c-94b3-8bfc3f289e44">

Reply:
Cool! Let me know when you need another review. This looks good though (I'm glad you're getting a chance to learn more about PID by messing with this as well)